hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactionPlanOperator.java

```diff
@@ -82,7 +83,7 @@ public void open() throws Exception {
     // when starting up, rolls back all the inflight compaction instants if there exists,
     // these instants are in priority for scheduling task because the compaction instants are
     // scheduled from earliest(FIFO sequence).
-    CompactionUtil.rollbackCompaction(table);
+    CompactionUtil.rollbackCompaction(table, FlinkWriteClients.createWriteClient(conf, getRuntimeContext()));
```

Comment:
Can we create a singleton write client for the whole operator and be caution that the write client has two kinds of instantces: server for coordinator and client for regular operators.